packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/bitmap.dart

```diff
@@ -171,8 +234,565 @@ class BitmapDescriptor {
+  /// [assetName] is the name of the asset to load.
+  /// [bundle] and [package] specify the asset's location if outside of the
+  /// default.
+  /// [width] and [height] can optionally control the dimensions of the rendered
+  /// image.
+  /// [imagePixelRatio] controls the scale of the image relative to the device's
+  /// pixel ratio. It defaults resolved asset image pixel ratio. The value is
+  /// ignored if [width] or [height] is provided.
+  ///
+  /// See [AssetMapBitmap.create] for more information on the parameters.
+  ///
+  /// Returns a Future that completes with a new [AssetMapBitmap] instance.
+  static Future<AssetMapBitmap> asset(
+    ImageConfiguration configuration,
+    String assetName, {
+    AssetBundle? bundle,
+    String? package,
+    double? width,
+    double? height,
+    double? imagePixelRatio,
+    MapBitmapScaling bitmapScaling = MapBitmapScaling.auto,
+  }) async {
+    return AssetMapBitmap.create(
+      configuration,
+      assetName,
+      bundle: bundle,
+      package: package,
+      width: width,
+      height: height,
+      imagePixelRatio: imagePixelRatio,
+      bitmapScaling: bitmapScaling,
+    );
+  }
+
+  /// Creates a [BitmapDescriptor] from byte data using [BytesMapBitmap].
+  ///
+  /// This method wraps [BytesMapBitmap] constructor for ease of use within the
+  /// context of creating [BitmapDescriptor] instances.
+  ///
+  /// [byteData] is the PNG-encoded image data.
+  /// [imagePixelRatio] controls the scale of the image relative to the device's
+  /// pixel ratio. It defaults to the natural resolution if not specified.
+  /// The value is ignored if [width] or [height] is provided.
+  /// [width] and [height] can optionally control the dimensions of the rendered
+  /// image.
+  ///
+  /// See [BytesMapBitmap] for more information on the parameters.
+  ///
+  /// Returns a new [BytesMapBitmap] instance.
+  static BytesMapBitmap bytes(
+    Uint8List byteData, {
+    double? imagePixelRatio,
+    double? width,
+    double? height,
+    MapBitmapScaling bitmapScaling = MapBitmapScaling.auto,
+  }) {
+    return BytesMapBitmap(
+      byteData,
+      imagePixelRatio: imagePixelRatio,
+      width: width,
+      height: height,
+      bitmapScaling: bitmapScaling,
+    );
+  }
+
   final Object _json;
 
   /// Convert the object to a Json format.
   Object toJson() => _json;
 }
+
+/// Represents a [BitmapDescriptor] base class for map bitmaps.
+///
+/// See [AssetMapBitmap] and [BytesMapBitmap] for concrete implementations.
+abstract class MapBitmap extends BitmapDescriptor {
+  MapBitmap._({
+    required this.bitmapScaling,
+    required this.imagePixelRatio,
+    this.width,
+    this.height,
+  }) : super._(const <Object>[]);
+
+  /// The scaling method of the bitmap.
+  final MapBitmapScaling bitmapScaling;
+
+  /// The pixel ratio of the bitmap.
+  final double imagePixelRatio;
+
+  /// The target width of the bitmap in logical pixels.
```

Comment:
Could you add some detail about what happens if one or both of `width` and `height` are null, or if they are both non-null and the aspect ratio doesn't match (e.g., does it stretch, or fit within, or something else)?

Along the lines of what is on the later method, but on the public API surface rather than on a private method where clients are unlikely to see it.

Reply:
Updated documentation for the class and also for the instance variables